index.html

```diff
@@ -2,45 +2,199 @@
+            min-height: 100vh;
+        }
+
+        .choice-container {
+            display: inline-flex;
+            flex-flow: column;
+        }
+    </style>
 </head>
 
 <body>
-    <div style="margin-bottom: 1em;">
-        <a id="link"></a>
-        <button id="copyLink" onclick="copyLink()" style='display:none'>
-            Copy link
+
+    <!-- Text box -->
+    <div class="choice-container">
+        <label id="outline"
+            class="mdc-text-field mdc-text-field--outlined mdc-text-field--with-leading-icon mdc-text-field--with-trailing-icon">
+            <!-- Outline formatting and label for the text field -->
+            <span class="mdc-notched-outline">
+                <span class="mdc-notched-outline__leading"></span>
+                <span class="mdc-notched-outline__notch">
+                    <span class="mdc-floating-label" id="choice-label">Write word</span>
+                </span>
+                <span class="mdc-notched-outline__trailing"></span>
+            </span>
+
+            <!-- Language picker -->
+            <button class="mdc-text-field__icon mdc-text-field__icon--leading mdc-icon-button" style="margin-left: 0"
+                onclick="toggleLanguage()">
+                <img src="en.png" id="language" />
+            </button>
+            <!-- Actual input field -->
+            <input id="choice" type="text" class="mdc-text-field__input" aria-labelledby="choice-label" maxlength="5"
+                onkeyup="validateWord()" onchange="validateWord()">
+            <!-- Visibility button -->
+            <i id="visibility" class="material-icons mdc-text-field__icon mdc-text-field__icon--trailing" tabindex="0"
+                role="button" onclick="toggleVisibility()">visibility</i>
+        </label>
+        <div class="mdc-text-field-helper-line">
+            <!-- Report invalid word -->
+            <div id="error"
+                class="mdc-text-field-helper-text mdc-text-field-helper-text--persistent mdc-text-field-helper-text--validation-msg"
+                role="alert">
+            </div>
+            <!-- Report character count -->
+            <div class="mdc-text-field-character-counter">0 / 5</div>
+        </div>
+    </div>
+    <br>
+
+    <!-- Buttons for sharing, copying and follwing play link -->
+    <div>
+        <button class="mdc-icon-button material-icons" id="randomWord" onclick="pickRandomWord()"
+            style="margin-right: 2em;">
+            <div class="mdc-icon-button__ripple"></div>
+            shuffle
+        </button>
+        <button class="mdc-icon-button material-icons" onclick="copyLink()">
+            <div class="mdc-icon-button__ripple"></div>
+            content_copy
+        </button>
+        <button class="mdc-icon-button material-icons" onclick="shareLink()">
+            <div class="mdc-icon-button__ripple"></div>
+            share
+        </button>
+        <button class="mdc-icon-button material-icons" onclick="playWord()">
```

Comment:
Minor nitpick (can be updated later or not at all): I think that instead of having a button that calls a function that redirects you to a new url, you should just make this a `<a>` link with `href` that is updated whenever the word changes.

Reply:
I think I prefer a button

Reply:
See the other comment about why links are better at being links than buttons. But this a more complex problem where... I can be convinced to let it be. 🤷‍♂️
Because you actually do some work calculating the URL before redirecting.